Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs

```diff
@@ -64,47 +55,55 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
-        protected void Dispose(bool disposing)
+        protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
-                _cellularMonitor.Cancel();
-                _wifiMonitor.Cancel();
-                _wiredMonitor.Cancel();
-                _loopbackMonitor.Cancel();
-                _otherMonitor.Cancel();
-
-                _cellularMonitor.Dispose();
-                _wifiMonitor.Dispose();
-                _wiredMonitor.Dispose();
-                _loopbackMonitor.Dispose();
-                _otherMonitor.Dispose();
+                foreach (var monitor in _monitors)
+                {
+                    monitor.Cancel();
+                    monitor.Dispose();
+                }
+
+                _monitors.Clear();
             }
         }
 
-        private void UpdateWiredSnapshot(NWPath nWPath)
+        protected virtual IEnumerable<ConnectionType> FilterConnectionTypes(IList<NWInterfaceType> activeNetworkTypes)
         {
-            HandleUpdateSnapshot(nWPath, NWInterfaceType.Wired);
-        }
+            if (activeNetworkTypes.Contains(NWInterfaceType.Other) && activeNetworkTypes.Count > 1)
+            {
+                activeNetworkTypes.Remove(NWInterfaceType.Other);
+            }
 
-        private void UpdateWiFiSnapshot(NWPath nWPath)
-        {
-            HandleUpdateSnapshot(nWPath, NWInterfaceType.Wifi);
+            return activeNetworkTypes.Select(Convert);
         }
 
-        private void UpdateCeccularSnapshot(NWPath nWPath)
+        protected virtual ConnectionType Convert(NWInterfaceType networkType)
         {
-            HandleUpdateSnapshot(nWPath, NWInterfaceType.Cellular);
+            return networkType switch
+            {
+                NWInterfaceType.Cellular => ConnectionType.Cellular,
+                NWInterfaceType.Wifi => ConnectionType.WiFi,
+                _ => ConnectionType.Other
+            };
         }
 
-        private void UpdateLoopbackSnapshot(NWPath nWPath)
+        private NWPathMonitor RegisterMonitor(NWInterfaceType type)
         {
-            HandleUpdateSnapshot(nWPath, NWInterfaceType.Loopback);
+            _connectionStatuses.TryAdd(type, false);
+
+            var monitor = CreateMonitor(type, nWPath => HandleUpdateSnapshot(nWPath, type));
+            monitor.Start();
+            return monitor;
         }
 
-        private void UpdateOtherSnapshot(NWPath nWPath)
+        private NWPathMonitor CreateMonitor(NWInterfaceType type, Action<NWPath> action)
```

Comment:
Looks like this method is only used in `RegisterMonitor.` Maybe it makes sense to rename `RegisterMonitor` to something like `CreateAndRegisterMonitor` and get rid of `CreateMonitor` method completely?
I'm not really sure this will be better than it is right now, but this is just an idea. What do you think?

Reply:
`CreateMonitor` only creates the monitor, one responsibility (can be exported in the future as utility).

`RegisterMonitor` more complex method, describes the specific features of registration.

I think we should leave the methods separately.